### PR TITLE
US21 (correction)

### PR DESCRIPTION
**Pull request by austincarrjones**
## Describe Changes Made
Take 2! This should be full test and model test for Admin dash US21

## If PR is linked to a Issue, Add Issue
- Top Customers

## PR Checklist
- [X] Added Reviewer
- [X] Followed TDD, And Test are Passing
- [X] Linked Issues for Project Board

### CODE REVIEW

File: app/models/customer.rb [hunk 1]
```diff
@@ -2,4 +2,14 @@ class Customer < ApplicationRecord
   has_many :invoices
   has_many :transactions, through: :invoices
   has_many :merchants, through: :invoices
+
+  def self.top_five_customers
```
Comment (JCL461437) [inline app/models/customer.rb:6]: Nice use of the where Austin!

File: spec/features/admin/index_spec.rb [hunk 2]
```diff
@@ -16,6 +16,46 @@
...
+      transaction_4 = Transaction.create!(invoice_id: invoice_4.id, result: 0)
+      transaction_5 = Transaction.create!(invoice_id: invoice_5.id, result: 0)
+      transaction_6 = Transaction.create!(invoice_id: invoice_6.id, result: 0)
+      transaction_7 = Transaction.create!(invoice_id: invoice_7.id, result: 0)
+      transaction_8 = Transaction.create!(invoice_id: invoice_8.id, result: 0)
+      transaction_9 = Transaction.create!(invoice_id: invoice_9.id, result: 1)
+
+      visit "/admin"
+      
+      expect(Customer.count).to eq(6)
```
Comment (JCL461437) [inline spec/features/admin/index_spec.rb:50]: This test is much more robust. Great work.

File: spec/models/customer_spec.rb [hunk 3]
```diff
@@ -6,4 +6,49 @@
...
+      transaction_2 = Transaction.create!(invoice_id: invoice_2.id, result: 0)
+      transaction_3 = Transaction.create!(invoice_id: invoice_3.id, result: 0)
+      transaction_4 = Transaction.create!(invoice_id: invoice_4.id, result: 0)
+      transaction_5 = Transaction.create!(invoice_id: invoice_5.id, result: 0)
+      transaction_6 = Transaction.create!(invoice_id: invoice_6.id, result: 0)
+      transaction_7 = Transaction.create!(invoice_id: invoice_7.id, result: 0)
+      transaction_8 = Transaction.create!(invoice_id: invoice_8.id, result: 0)
+      transaction_9 = Transaction.create!(invoice_id: invoice_9.id, result: 1)
+      
+      top_customers = Customer.top_five_customers
```
Comment (JCL461437) [inline spec/models/customer_spec.rb:39]: That is a very clever way to make the test data.

**Review by JCL461437 — APPROVED** [review]
Great work Austin!

**Review by GBowman1 — APPROVED** [review]

**Pull request merged**